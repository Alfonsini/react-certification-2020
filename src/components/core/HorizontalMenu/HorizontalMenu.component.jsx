import React from 'react';

import { StyledUl, StyledLi, StyledLink, StyledLinkTitle } from './HorizontalMenu.styles';

function MenuHorizontal() {
  return (
    <StyledUl role="menu">
      <StyledLi>
        <StyledLink to="/" className="disabledCursor">
          <StyledLinkTitle textTruncate>Home</StyledLinkTitle>
        </StyledLink>
      </StyledLi>
      <StyledLi>
        <StyledLink to="/" className="disabledCursor">
          <StyledLinkTitle textTruncate>Favorites</StyledLinkTitle>
        </StyledLink>
      </StyledLi>
    </StyledUl>
  );
}

export default MenuHorizontal;
