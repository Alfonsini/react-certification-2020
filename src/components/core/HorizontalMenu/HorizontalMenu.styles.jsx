import styled from 'styled-components';

const StyledUl = styled.ul`
  list-style-type: none;
  display: none;
  padding: 0;

  @media screen and (max-width: 500px) {
    display: block;
    flex-direction: row;
    justify-content: flex-end;
  }
`;

const StyledLi = styled.li`
  text-align: center;
  margin: 15px auto;

  @media screen and (max-width: 500px) {
    margin: 0;
  }
`;

const StyledLink = styled.a`
  text-decoration: none;

  @media screen and (max-width: 500px) {
    margin: 0;
  }
`;

const StyledLinkTitle = styled.span`
  ${(props) =>
    props.textTruncate &&
    `
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `}
`;

export { StyledUl, StyledLi, StyledLink, StyledLinkTitle };
