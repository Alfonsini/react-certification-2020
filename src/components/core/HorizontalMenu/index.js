export { default } from './HorizontalMenu.component';
