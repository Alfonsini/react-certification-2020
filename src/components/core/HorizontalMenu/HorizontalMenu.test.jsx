import React from 'react';
import { render, screen } from '@testing-library/react';

import HorizontalMenu from '.';

beforeEach(() => {
  render(<HorizontalMenu />);
});

afterEach(() => {});

describe('HorizontalMenu', () => {
  test('render correctly', () => {
    const menu = screen.getByRole('menu', { hidden: true });
    expect(menu).toMatchSnapshot();
  });

  test('home is available', () => {
    const option = screen.getByText('Home');
    expect(option).toMatchSnapshot();
  });

  test('favorites is available', () => {
    const option = screen.getByText('Favorites');
    expect(option).toMatchSnapshot();
  });
});
