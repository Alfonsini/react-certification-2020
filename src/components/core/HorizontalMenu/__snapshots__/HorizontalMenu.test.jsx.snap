// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalMenu favorites is available 1`] = `
<span
  class="sc-hKgILt fNqzIf"
>
  Favorites
</span>
`;

exports[`HorizontalMenu home is available 1`] = `
<span
  class="sc-hKgILt fNqzIf"
>
  Home
</span>
`;

exports[`HorizontalMenu render correctly 1`] = `
<ul
  class="sc-bdfBwQ cJHRYr"
  role="menu"
>
  <li
    class="sc-gsTCUz fXCZoy"
  >
    <a
      class="sc-dlfnbm gWYOlF disabledCursor"
      to="/"
    >
      <span
        class="sc-hKgILt fNqzIf"
      >
        Home
      </span>
    </a>
  </li>
  <li
    class="sc-gsTCUz fXCZoy"
  >
    <a
      class="sc-dlfnbm gWYOlF disabledCursor"
      to="/"
    >
      <span
        class="sc-hKgILt fNqzIf"
      >
        Favorites
      </span>
    </a>
  </li>
</ul>
`;
